flutter_cache_manager/lib/src/cache_managers/image_cache_manager.dart

```diff
@@ -131,3 +132,13 @@ mixin ImageCacheManager on BaseCacheManager {
     }
   }
 }
+
+Future<ui.Image> _decodeImage(File file, {int? width, int? height, bool allowUpscaling = false}) {
+  // since we set allowUpscaling to false, the width and height parameters are actually max-width.
+  final image = ResizeImage(FileImage(file), width: width, height: height, allowUpscaling: allowUpscaling);
```

Comment:
In line 76 you call this without any width or height and that's not allowed for ResizeImage
```suggestion
  var shouldResize = width != null || height != null;
  var fileImage = FileImage(file);
  final image = shouldResize
      ? ResizeImage(fileImage,
          width: width, height: height, allowUpscaling: allowUpscaling)
      : fileImage as ImageProvider;
```